File: backend/src/main/java/de/interaapps/pastefy/model/requests/auth/InteraAppsExternalAccessRequest.java
package de.interaapps.pastefy.model.requests.auth;

import java.util.List;

public class InteraAppsExternalAccessRequest {
    public String provider;
    public String appId;
    public String appSecret;
    public String scope;

    public List<String> scopeList;
    public List<String> appScopeList;
    public String userId;
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/dto/auth/InteraAppsExternalAccessRequest.kt
package de.interaapps.pastefy.dto.auth

import jakarta.validation.constraints.NotBlank

data class InteraAppsExternalAccessRequest(
    val provider: String? = null,
    @field:NotBlank val appId: String,
    @field:NotBlank val appSecret: String,
    val scope: String? = null,
    val scopeList: List<String> = emptyList(),
    val appScopeList: List<String> = emptyList(),
    @field:NotBlank val userId: String,
)
